packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewHostApi.m

```diff
@@ -0,0 +1,176 @@
+// found in the LICENSE file.
+
+#import "FWFWebViewHostApi.h"
+#import "FWFDataConverters.h"
+
+@implementation FWFWebView
+- (void)setFrame:(CGRect)frame {
+  [super setFrame:frame];
+  // Prevents the contentInsets to be adjusted by iOS and gives control to Flutter.
+  self.scrollView.contentInset = UIEdgeInsetsZero;
+  if (@available(iOS 11, *)) {
+    // Above iOS 11, adjust contentInset to compensate the adjustedContentInset so the sum will
+    // always be 0.
+    if (UIEdgeInsetsEqualToEdgeInsets(self.scrollView.adjustedContentInset, UIEdgeInsetsZero)) {
+      return;
+    }
+    UIEdgeInsets insetToAdjust = self.scrollView.adjustedContentInset;
+    self.scrollView.contentInset = UIEdgeInsetsMake(-insetToAdjust.top, -insetToAdjust.left,
+                                                    -insetToAdjust.bottom, -insetToAdjust.right);
+  }
+}
+
+- (nonnull UIView *)view {
+  return self;
+}
+@end
+
+@interface FWFWebViewHostApiImpl ()
+@property(nonatomic) FWFInstanceManager *instanceManager;
+@end
+
+@implementation FWFWebViewHostApiImpl
+- (instancetype)initWithInstanceManager:(FWFInstanceManager *)instanceManager {
+  self = [self init];
+  if (self) {
+    _instanceManager = instanceManager;
+  }
+  return self;
+}
+
+- (FWFWebView *)getWebViewInstance:(NSNumber *)instanceId {
+  return (FWFWebView *)[self.instanceManager instanceForIdentifier:instanceId.longValue];
+}
+
+- (void)webViewWithInstanceId:(nonnull NSNumber *)instanceId
+                  loadRequest:(nonnull FWFNSUrlRequestData *)request
+                        error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  NSURLRequest *urlRequest = FWFNSURLRequestFromRequestData(request);
+  if (!urlRequest) {
+    *error = [FlutterError errorWithCode:@"CreateNSURLRequestFailure"
+                                 message:@"Failed instantiating an NSURLRequest."
+                                 details:[NSString stringWithFormat:@"Url was: '%@'", request.url]];
+    return;
+  }
+  [[self getWebViewInstance:instanceId] loadRequest:urlRequest];
+}
+
+- (void)webViewWithInstanceId:(nonnull NSNumber *)instanceId
+           setCustomUserAgent:(nullable NSString *)userAgent
+                        error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  [[self getWebViewInstance:instanceId] setCustomUserAgent:userAgent];
+}
+
+- (nullable NSNumber *)
+    webViewWithInstanceIdCanGoBack:(nonnull NSNumber *)instanceId
+                             error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  return @([self getWebViewInstance:instanceId].canGoBack);
+}
+
+- (nullable NSString *)
+    urlForWebViewWithInstanceId:(nonnull NSNumber *)instanceId
+                          error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  return [self getWebViewInstance:instanceId].URL.absoluteString;
+}
+
+- (nullable NSNumber *)canGoForwardInstanceId:(nonnull NSNumber *)instanceId
```

Comment:
We should fix all the names now, even the ones you aren't implementing yet, since this is the PR that's creating the generated API and the stubs for them. So this should match whatever is done for canGoBack.